### PR TITLE
feat: simulate token drawer (PIN-9879)

### PR DESCRIPTION
**Pull request by l-giacomini**
## Issue
- [PIN-9879](https://pagopa.atlassian.net/browse/PIN-9879)
## Description / Context / Why
- Updated general form drawer rendering logic for InterationType: "ASYNC" and memberType: "PRODUCER" in order to show the following fields when open:
```
producerKeychainId
producerId
eserviceId
```
- Updated tests

[PIN-9879]: https://pagopa.atlassian.net/browse/PIN-9879?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/VoucherInstructionsGeneralFormCurrentIdsDrawer.tsx [hunk 1]
```diff
@@ -99,6 +110,39 @@ export const VoucherInstructionsGeneralFormCurrentIdsDrawer: React.FC<
...
+            <InformationContainer
+              direction="column"
+              label={'producerId'}
+              labelDescription={t('producerId')}
+              copyToClipboard={{ value: keychain.producer.id }}
+              content={keychain.producer.id}
+            />
+          </>
+        )}
+        {eserviceId && (
```
Comment (Alepazz) [inline src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/VoucherInstructionsGeneralFormCurrentIdsDrawer.tsx]: The drawer now renders `label="eserviceId"` in two places: once inside the existing `{purpose && (...)}` block (with `content={purpose.eservice.id}`) and once here (with `content={eserviceId}`). When the form has both `purposeId` and `eserviceId` set, the user sees two `eserviceId` rows — possibly with different values. In the current flows the two should be mutually exclusive (Consumer uses `purposeId`, Producer uses `eserviceId`), but that depends on the form clearing unused fields when switching modes. Worth either gating explicitly with `{eserviceId && !purpose && (...)}` or confirming the form-level mutual exclusion.

File: src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/VoucherInstructionsGeneralFormCurrentIdsDrawer.tsx [hunk 2]
```diff
@@ -71,14 +79,17 @@ export const VoucherInstructionsGeneralFormCurrentIdsDrawer: React.FC<
...
-
-        <InformationContainer
-          direction="column"
-          label="clientId"
-          labelDescription={t('clientIdDescription')}
-          copyToClipboard={{ value: clientId }}
-          content={clientId}
-        />
+        {clientId && (
+          <>
```
Comment (Alepazz) [inline src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/VoucherInstructionsGeneralFormCurrentIdsDrawer.tsx]: Fragment wrapping a single child is unnecessary here. Same pattern in the new `producerKeychainId`, `keychain`, and `eserviceId` blocks below.

File: src/static/locales/en/voucher.json [hunk 3]
```diff
@@ -123,7 +123,10 @@
       "consumerOrganizationId": {
         "label": "organizationId (consumer)",
         "description": "The ID of your party"
-      }
+      },
+      "eserviceId": "The ID of the selected e-service",
```
Comment (Alepazz) [inline src/static/locales/en/voucher.json]: Inconsistent shape with the keys above (e.g. `consumerOrganizationId: { label, description }`). The new keys are flat strings used only for `labelDescription`, while the `label` is hardcoded in the component. If at some point the labels need translation too, this will require restructuring. Consider matching the existing nested shape now:
```json
"eserviceId": {
  "label": "eserviceId",
  "description": "The ID of the selected e-service"
}
```
Same for `producerKeychainId` and `producerId` (and the IT counterpart).

File: src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/VoucherInstructionsGeneralFormCurrentIdsDrawer.tsx [hunk 4]
```diff
@@ -99,6 +110,39 @@ export const VoucherInstructionsGeneralFormCurrentIdsDrawer: React.FC<
...
+              copyToClipboard={{ value: producerKeychainId }}
+              content={producerKeychainId}
+            />
+          </>
+        )}
+        {keychain && (
+          <>
+            <InformationContainer
+              direction="column"
+              label={'producerId'}
```
Comment (Alepazz) [inline src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/VoucherInstructionsGeneralFormCurrentIdsDrawer.tsx]: Inconsistent quoting style with the surrounding `label="producerKeychainId"` / `label="eserviceId"` (no braces, double quotes).
```suggestion
              label="producerId"
```

File: src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/__tests__/VoucherInstructionsGeneralFormCurrentIdsDrawer.test.tsx [hunk 5]
```diff
@@ -10,7 +10,7 @@ vi.mock('@/api/purpose', () => ({
       queryFn: async () => ({
         id: 'purpose-1',
         eservice: {
-          id: 'eservice-1',
+          id: 'purpose-eservice-1',
```
Comment (Alepazz) [inline src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/__tests__/VoucherInstructionsGeneralFormCurrentIdsDrawer.test.tsx:15]: Renaming the mock from `eservice-1` to `purpose-eservice-1` differentiates it from the new `eserviceId: 'eservice-1'` prop in the test, but sidesteps the underlying question: in production both values can coexist (see the duplication concern on the source file). The new test could assert that both `purpose-eservice-1` and `eservice-1` are rendered — or, if mutual exclusion is the intended contract, that only one is — to lock the intended behavior.

File: src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/__tests__/VoucherInstructionsGeneralFormCurrentIdsDrawer.test.tsx [hunk 6]
```diff
@@ -68,4 +84,18 @@ describe('VoucherInstructionsGeneralFormCurrentIdsDrawer', () => {
     expect(await screen.findByText('producer-1')).toBeInTheDocument()
     expect(await screen.findByText('consumer-1')).toBeInTheDocument()
   })
+
+  it('renders clientId, keychain and eserviceId props', async () => {
```
Comment (Alepazz) [inline src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/__tests__/VoucherInstructionsGeneralFormCurrentIdsDrawer.test.tsx]: Consider adding a negative-case test where `eserviceId` and `producerKeychainId` are empty strings — the three new `InformationContainer`s should not render. This would lock the gating logic added in the source file and prevent regressions if the conditions change.

**Review by Alepazz — COMMENTED** [review]
The PR is solid and well-tested. A few things worth evaluating below.

File: src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/VoucherInstructionsGeneralFormCurrentIdsDrawer.tsx (hunk 1 above)
Comment (l-giacomini) [reply src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/VoucherInstructionsGeneralFormCurrentIdsDrawer.tsx]: Fixed in: https://github.com/pagopa/pdnd-interop-frontend/pull/1877/changes/f0aa51bba0f64bfae545da5b557252606b24f316

File: src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/VoucherInstructionsGeneralFormCurrentIdsDrawer.tsx (hunk 2 above)
Comment (l-giacomini) [reply src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/VoucherInstructionsGeneralFormCurrentIdsDrawer.tsx]: Fixed in: https://github.com/pagopa/pdnd-interop-frontend/pull/1877/changes/f95fda9a10c693a54bdfdc6491f235664a3a6cca

File: src/static/locales/en/voucher.json (hunk 3 above)
Comment (l-giacomini) [reply src/static/locales/en/voucher.json]: Fixed in: https://github.com/pagopa/pdnd-interop-frontend/pull/1877/changes/281cda0b9da7bc584c8db7380509031670d26e01

File: src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/VoucherInstructionsGeneralFormCurrentIdsDrawer.tsx (hunk 4 above)
Comment (l-giacomini) [reply src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/VoucherInstructionsGeneralFormCurrentIdsDrawer.tsx]: Fixed in: https://github.com/pagopa/pdnd-interop-frontend/pull/1877/changes/281cda0b9da7bc584c8db7380509031670d26e01

File: src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/__tests__/VoucherInstructionsGeneralFormCurrentIdsDrawer.test.tsx (hunk 5 above)
Comment (l-giacomini) [reply src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/__tests__/VoucherInstructionsGeneralFormCurrentIdsDrawer.test.tsx:15]: Fixed in: https://github.com/pagopa/pdnd-interop-frontend/pull/1877/changes/f0aa51bba0f64bfae545da5b557252606b24f316

File: src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/__tests__/VoucherInstructionsGeneralFormCurrentIdsDrawer.test.tsx (hunk 6 above)
Comment (l-giacomini) [reply src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/__tests__/VoucherInstructionsGeneralFormCurrentIdsDrawer.test.tsx]: Fixed in: https://github.com/pagopa/pdnd-interop-frontend/pull/1877/changes/b6ba700713d6d960a73649e8fabe2976f32f6887

File: src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/__tests__/VoucherInstructionsGeneralFormCurrentIdsDrawer.test.tsx [hunk 7]
```diff
@@ -10,62 +12,105 @@ vi.mock('@/api/purpose', () => ({
...
+    renderWithApplicationContext(
+      <VoucherInstructionsGeneralFormCurrentIdsDrawer {...defaultProps} />,
+      { withReactQueryContext: true }
+    )
 
     expect(await screen.findByText('client-1')).toBeInTheDocument()
     expect(await screen.findByText('purpose-1')).toBeInTheDocument()
     expect(await screen.findByText('agreement-1')).toBeInTheDocument()
     expect(await screen.findByText('producer-1')).toBeInTheDocument()
     expect(await screen.findByText('consumer-1')).toBeInTheDocument()
```
Comment (Alepazz) [inline src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/__tests__/VoucherInstructionsGeneralFormCurrentIdsDrawer.test.tsx:75]: The default props include both `purposeId` and `eserviceId`, but the test doesn't assert that the `!purposeId` gate hides the eserviceId prop row. Adding `expect(screen.queryByText('eservice-1')).not.toBeInTheDocument()` (and asserting `purpose-eservice-1` is rendered) would lock the mutual-exclusion contract.

File: src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/VoucherInstructionsGeneralFormCurrentIdsDrawer.tsx [hunk 8]
```diff
@@ -99,6 +108,33 @@ export const VoucherInstructionsGeneralFormCurrentIdsDrawer: React.FC<
...
+        {producerKeychainId && (
+          <InformationContainer
+            direction="column"
+            label={t('producerKeychainId.label')}
+            labelDescription={t('producerKeychainId.description')}
+            copyToClipboard={{ value: producerKeychainId }}
+            content={producerKeychainId}
+          />
+        )}
+        {keychain && (
```
Comment (Alepazz) [inline src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/VoucherInstructionsGeneralFormCurrentIdsDrawer.tsx]: This block renders even if `keychain.producer.id` is empty — the InformationContainer would show a row with empty content. Symmetric with the other gates, `{keychain?.producer?.id && (...)}` would be safer. Edge case, but worth a thought.

File: src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/VoucherInstructionsGeneralFormCurrentIdsDrawer.tsx [hunk 9]
```diff
@@ -99,6 +108,33 @@ export const VoucherInstructionsGeneralFormCurrentIdsDrawer: React.FC<
             />
           </>
         )}
+        {producerKeychainId && (
+          <InformationContainer
+            direction="column"
+            label={t('producerKeychainId.label')}
```
Comment (Alepazz) [inline src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/VoucherInstructionsGeneralFormCurrentIdsDrawer.tsx:114]: The new fields use `label={t('...label')}` while the older blocks above (`clientId`, `purposeId`, `eserviceId`, etc.) use hardcoded `label="..."`. The translated value is the same string anyway, so it's a round-trip via i18n. Either keep the new ones hardcoded for consistency, or migrate the old ones to `t()` — pick one.

**Review by Alepazz — APPROVED** [review]
Nice work — previous review points well addressed. Left 3 inline suggestions, up to you whether to address them or not.

File: src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/__tests__/VoucherInstructionsGeneralFormCurrentIdsDrawer.test.tsx (hunk 7 above)
Comment (l-giacomini) [reply src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/__tests__/VoucherInstructionsGeneralFormCurrentIdsDrawer.test.tsx:75]: Fixed in: https://github.com/pagopa/pdnd-interop-frontend/pull/1877/changes/d4195408e97122492ac5c4a78d30d32a8ec26f39

File: src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/VoucherInstructionsGeneralFormCurrentIdsDrawer.tsx (hunk 8 above)
Comment (l-giacomini) [reply src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/VoucherInstructionsGeneralFormCurrentIdsDrawer.tsx]: Fixed in: https://github.com/pagopa/pdnd-interop-frontend/pull/1877/changes/9dd8ece5c5d5523a1808982b0780e88f989ec844

File: src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/VoucherInstructionsGeneralFormCurrentIdsDrawer.tsx (hunk 9 above)
Comment (l-giacomini) [reply src/pages/ConsumerSimulateGetVoucherPage/components/VoucherInstructions/VoucherInstructionsGeneralForm/VoucherInstructionsGeneralFormCurrentIdsDrawer.tsx:114]: Fixed in: https://github.com/pagopa/pdnd-interop-frontend/pull/1877/changes/1d5f91453a318f0c177b4d25d26ab431848f6742

**Review by borgesis95 — APPROVED** [review]
